bottlecap/src/otlp/agent.rs

```diff
@@ -0,0 +1,180 @@
+use datadog_trace_mini_agent::http_utils::{
+    log_and_create_http_response, log_and_create_traces_success_http_response,
+};
+use datadog_trace_utils::send_data::SendData;
+use datadog_trace_utils::trace_utils::TracerHeaderTags as DatadogTracerHeaderTags;
+use ddcommon::hyper_migration;
+use http_body_util::BodyExt;
+use hyper::{http, service::service_fn, Method, Response, StatusCode};
+use std::io;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use tokio::sync::mpsc::Sender;
+use tracing::{debug, error};
+
+use crate::{
+    config::Config, otlp::processor::Processor as OtlpProcessor, tags::provider,
+    traces::trace_processor::TraceProcessor,
+};
+
+const OTLP_TRACE_AGENT_PORT: usize = 4318;
```

Comment:
should this be configurable? Not sure what the standards are

Reply:
Doesn't seem like, our docs always recommend this port on localhost

Reply:
I asked, they do allow to configure any other port